src/odemis/driver/actuator.py

```diff
@@ -666,6 +667,41 @@ def _get_rot_matrix(self, invert=False):
             rotation *= -1
         return RigidTransform(rotation=rotation).matrix
 
+    def _getAxesRange(self):
+        """
+        Calculate the axes range for position VA based on the conversion matrix
+        from dep's positon.
+        """
+        # Dep's range
+        range_y = self.axes["y"].range
+        range_x = self.axes["x"].range
+
+        # Calculate rectangular border points on the given range
+        min_x = range_x[0]
+        max_x = range_x[1]
+        min_y = range_y[0]
+        max_y = range_y[1]
+        coordinates = [[min_x, min_y],
+                       [min_x, max_y],
+                       [max_x, min_y],
+                       [max_x, max_y]]
+
+        # Calculate range based on conversion matrix
+        new_coordinates = []
+        for val in coordinates:
+            vpos = self._convertPosFromdep(val)
+            new_coordinates.append(vpos)
+
+        # Update default axes range and increase by buffer
+        # to account for movements at the extreme location
+        new_range = numpy.array(new_coordinates, dtype=float)
+        buff = 2.e-03
```

Comment:
That's a quite arbitrary value! Why do you need extra room to handle movements near the end in the first place? Can you explain in the comment? 
Also, the value means that for now every axis get 4mm extra. Some hardware have a range of a couple of mm, so 4 mm is huge. Maybe add a percentage instead? (like 1%) Or can you completely remove this extra margin?

Reply:
Long story short, I think you can pass the range just as it is converted, without "buffer".